src/main/java/com/solace/samples/java/patterns/DirectReplierNonBlocking.java

```diff
@@ -0,0 +1,122 @@
+        messagingService.connect();  // blocking connect to the broker
+
+        //4. Register event handlers and callbacks for connection error handling.
+        setupConnectivityHandlingInMessagingService(messagingService);
+
+        //5. Build and start the Receiver object
+        RequestReplyMessageReceiver requestReplyMessageReceiver = messagingService.requestReply().createRequestReplyMessageReceiverBuilder().build(TopicSubscription.of(TOPIC_PREFIX + "*/direct/request"));
+        requestReplyMessageReceiver.start();
+        //5-A. Set up an event handler for situations where the reply message could not be published.
+        requestReplyMessageReceiver.setReplyFailureListener(failedReceiveEvent -> System.out.println("### FAILED RECEIVE EVENT " + failedReceiveEvent));
+
+        //6. Create an OutboundMessageBuilder for building the outbound reply message
+        final OutboundMessageBuilder outboundMessageBuilder = messagingService.messageBuilder();
+
+        //7. Define the handler for the incoming message.
+        final RequestReplyMessageReceiver.RequestMessageHandler messageHandler = (inboundMessage, replier) -> {
+
+            //This SOP is just for demo purposes, ideally considering the slow nature of console I/O, any such action should be avoided in message processing
+//          System.out.println("The inbound message is : " + inboundMessage.dump()); // Enable this for learning purposes as it logs a String representation of the whole Message
+            final String stringPayload = inboundMessage.getPayloadAsString();
+            System.out.println("The converted message payload is : " + stringPayload);
+
+            //7-B. Create the outbound message payload.
+            final StringBuilder outboundMessageStringPayload = new StringBuilder().append("Hello! Here is a response to your message on topic : ").append(inboundMessage.getDestinationName()).append(" with incoming payload :").append(stringPayload);
+
+            //7-C. Create the outbound message with headers and payload.
+            final OutboundMessage outboundMessage = outboundMessageBuilder.build(outboundMessageStringPayload.toString());
+
+            //This SOP is just for demo purposes, ideally considering the slow nature of console I/O, any such action should be avoided in message processing
+            System.out.println("The outbound message is : " + outboundMessage.getPayloadAsString());
+
+            //7-D. Post the reply to the incoming message
+            replier.reply(outboundMessage);
+        };
+
+        //8. Loop to identify message discards or errors and terminate if required. This should be handled in a more resilient manner
+        System.out.println(API + " " + SAMPLE_NAME + " connected, and running.");
+        //8-A. Create a scheduler variable with the required configurations to process the message in
+        final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(2);
```

Comment:
any significance to the choice of 2 for the `corePoolSize` here?

Reply:
- @Mrc0113 : The corePoolSize refers to the number of threads that are left behind in the pool even if they are idle. I configured it like this just as a test example. Ideally, this number is setup based on factors like application memory size configs, ingress and outgress rates, the time of processing etc. The development team has to do detailed load and performance testing to make sure that the config is right for their app and infra.

- @TamimiGitHub : What the shutdown means is no more tasks will be allocated to the executorService after its called and instance will be eligible for garbage collection when the JVM sees appropriate. Usually when this piece of code is embedded within an enterprise grade app based on spring boot etc, there are api hooks that can be used to ensure that the executor service is shutdown gracefully. This is quite a standard practise to implement graceful shutdown when you create executorServices manually. I really dont think there is a need to go this much in detail about thread handling basics in our samples. 
- The method that is being used is a more sophisticated method which will give more control to developers on the performance and handling of the threads. They can simply use the normal receiveAsync method which will just spawn an ExecutorService internally without any optimization or control.